### ISSUE TITLE
Incorrect output for complex gcc commands

### ISSUE THREAD

**Issue by coderkd10** [idank/explainshell#194]
when I enter the following command (a complex `gcc` compilation command) into explainshell it cause it to behave very weirdly and it generate very bad (partially wrong) explanation.

command : 
`gcc -xc++-header -I"../t4/thirdparty/md5" -I"../../PDFViewer/RMSDK/xml/uft/public" a.c`

url :
https://explainshell.com/explain?cmd=gcc+-xc%2B%2B-header+-I%22..%2Ft4%2Fthirdparty%2Fmd5%22+-I%22..%2F..%2FPDFViewer%2FRMSDK%2Fxml%2Fuft%2Fpublic%22+a.c

I will try and look into the source and try to analyse its cause and submit a pull request if I am able to fix it.

**Comment by stoivo** [comment]
PR is very welcome

**Comment by idank** [comment]
Yeah, gcc is a mess. It has a weird man page and compared to other binaries, its command line formatting is complicated that it doesn't bode well with explainshell trying to be as general as possible.

**Comment by mnciitbhu** [comment]
Simple gcc commands are also not working `g++ -std=c++17 main.cpp`. check [explainshell](https://explainshell.com/explain?cmd=g%2B%2B+-std%3Dc%2B%2B17+main.cpp)
